### PR TITLE
Add signing of outgoing envelope

### PR DESCRIPTION
**Pull request by fabiancook**

### CODE REVIEW

**Comment by fabiancook** [comment]
This is the start of a few commits, starting the discussion.

**Comment by fabiancook** [comment]
@jsdevel have any opinions for naming? Should 'signing' be something else, signing is only related if you are wanting to sign the request, but they could just be interceptors in general

File: Readme.md [hunk 1]
```diff
@@ -206,6 +206,23 @@ terminated.
   };
 ```
 
+### Server.*addSigner*(signer) - use the specified function to sign the response
+
+With a signer you can intercept the response envelope before it is sent to the client.
```
Comment (jsdevel) [inline Readme.md]: should this say `request envelope` instead?

File: lib/client.js [hunk 2]
```diff
@@ -21,10 +23,50 @@ var Client = function(wsdl, endpoint, options) {
...
+  this._signers.push(signer);
+};
+
+/**
+* Sign an envelope
+* @param {String} envelope The envelope to sign
+* @param {ClientEnvelopeSignerOptions} options Options to pass to the signer
+* @param {NodeCallback} callback Function to be called once the envelope is signed
+*/
+Client.prototype._sign = function(envelope, options, callback) {
```
Comment (jsdevel) [inline lib/client.js]: Is it possible to avoid this method altogether and just call `sign` directly?

File: lib/server.js [hunk 3]
```diff
@@ -72,73 +75,151 @@ Server.prototype._initializeOptions = function(options) {
   this.wsdl.options.attributesKey = options.attributesKey || 'attributes';
 };
 
+/**
+* @callback NodeCallback
+* @param {?*} error
+* @param {String} envelope
+*/
+
+/**
```
Comment (jsdevel) [inline lib/server.js]: I feel the jsdocs here are overkill.

Comment (fabiancook) [inline lib/server.js]: Fair enough, was following suit

File: lib/client.js (hunk 2 above)
Comment (fabiancook) [inline lib/client.js]: Yeah I guess, that wouldn't be a problem. Only thought was that you could have multiple. Will change to single

**Comment by fabiancook** [comment]
Think I am having a derp moment and buggered my fork. Lets try again

**Pull request closed without merging**